native/src/modules/app/components/Navigator.js

```diff
@@ -123,18 +126,52 @@ const Navigator = (props: PropsType) => {
     initialize().catch(error => setErrorMessage(error.message))
   }, [])
 
+  // The following is used to have correct mapping from categories route mapping in redux state to the actual routes
+  useEffect(() => {
```

Comment:
pew can't we just fetch the new categories in when pressing the button on the landing page? For what did you use the key variable previously? Can we maybe just move the key to the params of a route?

Reply:
We need the key of the route. Otherwise the mapping does not work.

Reply:
Moving the key to params does not work as the key is already a property of the route property and would therefore lead to way more confusion. This whole workaround is needed because the ability to replace a route with a route with predefined key was removed.